### PR TITLE
Replace Travis CI build with GitHub Action

### PR DESCRIPTION
**Pull request by mvz**
travis-ci.org is shutting down and Open Source support will be severely limited on travis-ci.com. Build with GitHub Actions instead.

Differences with the existing build:
- Added 3.0
- Removed ruby-head. GitHub Actions always fail when one of the builds fails.

### CODE REVIEW

**Comment by satoryu** [comment]
@mvz thank you for opening this PR. 

> Open Source support will be severely limited on travis-ci.com. 

Could you share any document about this limitation?

**Comment by mvz** [comment]
Basically https://news.ycombinator.com/item?id=25338983 and the resources linked there.

**Comment by satoryu** [comment]
@mvz Thank you for sharing the doc. 
I've read it and [this doc](https://docs.travis-ci.com/user/billing-overview/#credits-vs-oss-only-credits)
I see why this PR is needed. Thanks

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    branches: [ master ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        ruby: [2.5, 2.6, 2.7, "3.0"]
```
Comment (satoryu) [inline .github/workflows/ruby.yml:20]: I have one question: why is only `3.0` surrounded with `"`?

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# run tests with Rake
+# For more information see: https://github.com/marketplace/actions/setup-ruby-jruby-and-truffleruby
+
+name: Ruby
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (satoryu) [inline .github/workflows/ruby.yml:11]: These `branches` means that the jobs run only for updates on the branch `master`. 
So these `branches` are not needed, I think.

**Review by satoryu — CHANGES_REQUESTED** [review]

File: .github/workflows/ruby.yml (hunk 1 above)
Comment (mvz) [reply .github/workflows/ruby.yml:20]: This is to make the Ruby version appear as 3.0 in the build results. GitHub Actions displays the other numbers correctly automatically.

Compare the left-hand side of this display: https://github.com/mvz/gir_ffi/actions/runs/476007991

With this one: https://github.com/mvz/happymapper/runs/1667668414

I suppose you could surround all of them with quotes to be consistent.

File: .github/workflows/ruby.yml (hunk 2 above)
Comment (satoryu) [reply .github/workflows/ruby.yml:11]: Ah, I misunderstood this settings. It works as we expected.

**Comment by satoryu** [comment]
@mvz please resolve the conflicts on `.travis.yml`. The file can be removed.

**Comment by mvz** [comment]
@satoryu I rebased and resolved the conflict.

**Review by satoryu — APPROVED** [review]

**Pull request merged**